### PR TITLE
[MAINT] Handle deprecations in dependencies that raise warnings in pre-release CI test

### PR DESCRIPTION
**Pull request by ymzayek**
This is just to try to get ahead of potential failures and useful if we want to implement #3541

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM.

**Comment by ymzayek** [comment]
In this PR I did not handle internal deprecation warnings or `PendingDeprecationWarning`s. There are 2 matplotlib deprecations I'm trying to handle but these needs a bit more care as to not change the default plots. Any input on these would be helpful:

```bash
nilearn/plotting/tests/test_surf_plotting.py: 23 warnings
  /home/runner/work/nilearn/nilearn/nilearn/plotting/surf_plotting.py:1305: MatplotlibDeprecationWarning:
  
  The dist attribute was deprecated in Matplotlib 3.6 and will be removed two minor releases later.
```
```bash
nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py::test_plot_carpet_with_atlas
  /home/runner/work/nilearn/nilearn/nilearn/plotting/img_plotting.py:1464: MatplotlibDeprecationWarning:
  
  Auto-removal of overlapping axes is deprecated since 3.6 and will be removed two minor releases later; explicitly call ax.remove() as needed.
```

**Comment by ymzayek** [comment]
Also, forgot to mention that I'm now using the package `packaging` in `nilearn.version.py` which is not a dependency of nilearn but is required by nibabel

**Comment by ymzayek** [comment]
When resolving the conflicts I removed handling of `pytest.warns(None)`. Will redo asap

File: setup.cfg [hunk 1]
```diff
@@ -14,6 +14,7 @@ install_requires =
 	requests>=2.25.0
 	scikit-learn>=1.0.0
 	scipy>=1.6.0
+    packaging
```
Comment (ymzayek) [inline setup.cfg:17]:
```suggestion
	scipy>=1.6.0
        packaging
```

**Comment by ymzayek** [comment]
Ok merging

**Pull request merged**